internal/api/describe.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	"github.com/edulinq/autograder/internal/api/core"
+)
+
+// Routes must be validated before calling Describe.
```

Comment:
Am I missing something, I don't see any Validate() function.

This is a little tricky.
The core of this is in NewAPIRoute(), where we expect to be able to get the types.
So a route is effectively "validated" there, which is fine.
But an incorrectly formatted API handler can cause us to log an error (but not actually return one).
Then that would route would cause issues here.
Note that then the API endpoint will also not validate, so this is something we can guarantee will be caught in tests.

Maybe we need to change NewAPIRoute() to MustNewAPIRoute()?
Can you verify that this would be safe?